lib/packages_file.dart

```diff
@@ -106,10 +106,13 @@ Map<String, Uri> parse(List<int> source, Uri baseLocation,
 /// If [baseUri] is provided, package locations will be made relative
 /// to the base URI, if possible, before writing.
 ///
+/// If [defaultPackageName] is provided, the output will contain a
+/// `:<defaultPackageName>` entry.
```

Comment:
If defaultPackageName is provided, then the map must not contain an empty-named entry, otherwise it can (and we should probably check that it has a valid package name as value), and then that will be the default package name.

That's the format you get when reading a `.packages` file, so we should support writing it too.